### PR TITLE
API changes and improvements

### PR DESCRIPTION
**Pull request by elcorto**
* API: Remove `c2st_()`, add `return_scores` kwd to `c2st()` instead
* API: Support passing `cv` and `clf` class instances, remove `clf_class` and
  `clf_kwargs`, retain previous defaults but now allow all classifiers and CV
  splitters with sklearn API to be passed
* ENH: Add more examples to README
* ENH: Add optional NaN handling
* ENH: Fix all typing issues, add `mypy.ini`
* TST: Add API-only tests
* BEH: Use `ubuntu-latest` in CI

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,27 +1,66 @@
...
+>>> rng=np.random.default_rng(seed=123)
+
+# same distribution (Gaussian N(0,1)), D=20, 1000 points each
+>>> X=rng.normal(loc=0, scale=1, size=(1000,20))
+>>> Y=rng.normal(loc=0, scale=1, size=(1000,20))
+>>> c2st(X, Y)
+0.4970122828225085
 
-The `compare` method returns (by default) the accuracy on how well a binary classifyer was able to classify Y from X while being trained on the concatenated dataset `(X,Y)`. All samples of `X` have received the label `0` and `Y` has received the label `1`.
+# now shift the mean of Y a bit
```
Comment (psteinb) [inline README.md]:
```suggestion
# now shift the mean of Y by 0.3
```

Comment (psteinb) [reply README.md]: Let's be precise and help people not familiar with the `np.random` API to understand the example.

File: README.md [hunk 2]
```diff
@@ -1,27 +1,66 @@
...
+>>> Y=rng.normal(loc=0.3, scale=1, size=(1000,20))
+0.6964673015530594
 
-## in `rsc`
+# and some more
```
Comment (psteinb) [inline README.md]:
```suggestion
# let's move Y more extensively away from X to 1.5
```

File: README.md [hunk 3]
```diff
@@ -1,27 +1,66 @@
...
+>>> Y=rng.normal(loc=1.5, scale=1, size=(1000,20))
+>>> c2st(X, Y)
+0.9994791666666666
 
-Small validation study which NN architecture exposes more utility for `c2st` two sample testing. At this point, the analysis you find in [rsc/c2st_results.ipynb](rsc/c2st_results.ipynb) has by far not any academic scrutiny as you'd find in Lopez-Paz et al. However, it can serve as guidance where which implementation of `c2st` can shine. 
+# or change the distribution's width
```
Comment (psteinb) [inline README.md]:
```suggestion
# or change the distribution's variance, so we compare a narrow to a wide distribution at the same loc
```

**Review by psteinb — CHANGES_REQUESTED** [review]

**Comment by elcorto** [comment]
I added the suggested README changes. @psteinb is there anything else to be done?

**Review by psteinb — APPROVED** [review]
LGTM!

**Comment by psteinb** [comment]
This is such a great addition. Thanks.

**Pull request merged**